pyprotista/parsers/ident/comet_2020_01_4_parser.py

```diff
@@ -149,25 +197,23 @@ def _map_mods_and_sequences(self):
                 .agg(";".join, axis=1)
                 .str.rstrip(";")
             )
-
-        modification_mass_map = {
-            sm.attrib["massDelta"]: sm.find(".//{*}cvParam[@cvRef='UNIMOD']").attrib[
-                "name"
-            ]
-            for sm in modifications
-        }
         lookup = {}
-        for pep in self.root.findall(".//{*}Peptide"):
-            id = pep.attrib.get("id", "")
-            lookup[id] = {"modifications": []}
-            lookup[id]["sequence"] = pep.find(".//{*}PeptideSequence").text
-            for child in pep.findall(".//{*}Modification"):
-                lookup[id]["modifications"].append(
-                    f"{modification_mass_map[child.attrib['monoisotopicMassDelta']]}:{child.attrib['location']}"
-                )
-            lookup[id]["modifications"] = ";".join(lookup[id]["modifications"])
+        for pep_sequence, attribs in peptide_lookup.items():
```

Comment:
maybe worth looking into getting mass_to_mod maps before reading mods from xml.
would allow us to assemble the mods correctly during the xml iteration and could be done as string instead of list.
unsure though.

Reply:
i will look into doing a separate xml iteration for that purpose.
won't work in one iteration, since get_peptide_lookup takes place before get_modifciation_mass_map.